File: mdm-server/src/main/java/com/edusphere/mdmserver/domain/command/dto/CommandDto.java
package com.edusphere.mdmserver.domain.command.dto;

import com.edusphere.mdmserver.domain.command.enums.CommandStatus;
import com.edusphere.mdmserver.domain.command.enums.CommandType;
import lombok.Builder;
import lombok.Data;

import java.time.Instant;
import java.util.Map;
import java.util.UUID;

@Data
@Builder
public class CommandDto {
    private UUID id;
    private String deviceId;
    private CommandType commandType;
    private Map<String, Object> payload;
    private CommandStatus status;
    private String errorMessage;
    private Instant createdAt;
    private Instant sentAt;
    private Instant acknowledgedAt;
    private Instant executedAt;
    private String createdBy;
}

File: android-agent/app/src/main/java/com/edusphere/agent/data/remote/model/CommandDto.kt
package com.edusphere.agent.data.remote.model

import com.google.gson.annotations.SerializedName

data class CommandDto(
    @SerializedName("id") val id: String,
    @SerializedName("deviceId") val deviceId: String,
    @SerializedName("commandType") val commandType: String, // LOCK_SCREEN, WIPE_DATA, RING_ALARM, etc.
    @SerializedName("payload") val payload: Map<String, Any>?,
    @SerializedName("status") val status: String,
    @SerializedName("errorMessage") val errorMessage: String?,
    @SerializedName("createdAt") val createdAt: String?,
    @SerializedName("sentAt") val sentAt: String?,
    @SerializedName("acknowledgedAt") val acknowledgedAt: String?,
    @SerializedName("executedAt") val executedAt: String?,
    @SerializedName("createdBy") val createdBy: String?
)
